tools/wptserve/wptserve/response.py

```diff
@@ -451,6 +451,11 @@ def write_headers(self, headers, status_code, status_message=None, stream_id=Non
         secondary_headers = []  # Non ':' prefixed headers are to be added afterwards
 
         for header, value in headers:
+            # h2_header by default is unicode in PY3
+            # header key on the other hand is bytes
+            header = self.decode(header)
+            if isinstance(value, binary_type):
+                value = self.decode(value)
             if header in h2_headers:
                 header = ':' + header
                 formatted_headers.append((header, str(value)))
```

Comment:
Do we still need to coerce `value` into string here given we've already decoded it (same in the else branch)?

Reply:
I think we might still need this. Value could be other types such as integer for "Content-length" header. Otherwise it will complain here [h2 file](https://github.com/web-platform-tests/wpt/blob/master/tools/third_party/h2/h2/utilities.py#L518).

Reply:
Thanks. Could you modify the docstring for this function to indicate that `value` in `headers` can be either string or int?